src/modules/rust.rs

```diff
@@ -127,24 +127,37 @@ fn find_rust_toolchain_file(context: &Context) -> Option<String> {
     // Look for 'rust-toolchain' as rustup does.
     // https://github.com/rust-lang/rustup.rs/blob/d84e6e50126bccd84649e42482fc35a11d019401/src/config.rs#L320-L358
 
-    fn read_first_line(path: &Path) -> Option<String> {
+    fn read_target(path: &Path) -> Option<String> {
         let content = fs::read_to_string(path).ok()?;
-        let line = content.lines().next()?;
-        Some(line.trim().to_owned())
+        let line = content.lines().next()?.trim();
+
+        if line.is_empty() || line == "[toolchain]" {
```

Comment:
I checked how rustup parses the file and they treat all files with no lines as invalid, a single line as the old format and anything longer as TOML ([source](https://github.com/rust-lang/rustup/blob/89912c4cf51645b9c152ab7380fd07574fec43a3/src/config.rs#L591-L616)). Would you mind changing the code to match rustup behaviour?

Reply:
Oh I see, I'll update it to stay as close as possible to rustup's implementation :+1: